src/pages/index.js

```diff
@@ -74,54 +97,76 @@ const cardTitleInput = addCardFormElement.querySelector(
 );
 const cardURLInput = addCardFormElement.querySelector(".modal__input_type_url");
 
-function closeWithEsc(event) {
-  if (event.key === "Escape") {
-    const modal = document.querySelector(".modal_opened");
-    closeModal(modal);
-  }
-}
-
-function closeModalOnRemoteClick(event) {
-  if (event.target === event.currentTarget) {
-    closeModal(event.currentTarget);
-  }
-}
-
-function openModal(modal) {
-  // add class to modal
-  document.addEventListener("keydown", closeWithEsc);
-  modal.addEventListener("mousedown", closeModalOnRemoteClick);
-  modal.classList.add("modal_opened");
-}
-
-function closeModal(modal) {
-  //remove class from modal
-  document.removeEventListener("keydown", closeWithEsc);
-  modal.removeEventListener("mousedown", closeModalOnRemoteClick);
-  modal.classList.remove("modal_opened");
-}
+const pictureURLInput = pictureFormElement.querySelector(
+  ".modal__input_type_url"
+);
 
 function renderCard(cardData) {
   const card = createCard(cardData);
   section.addItem(card);
-  //wrapper.prepend();
-  //Section.renderItems(item);
 }
 
 function handleImagePreview(cardData) {
-  popupImage.open(cardData); 
+  popupImage.open(cardData);
+}
+
+function handleSubmit(request, popupInstance, form, loadingText = "Saving...") {
+  popupInstance.renderLoading(true, loadingText);
+  // setTimeout(() => {
+  request()
+    .then(() => {
+      popupInstance.close();
+      if (form) {
+        form.disableButton();
+      }
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+  // }, 1000);
 }
 
 function handleProfileFormSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  editProfilePopup.close();
-  editProfileForm.reset();
+  function makeRequest() {
+    return api.setUserInfo(inputValues).then((userData) => {
+      userInfo.setUserInfo(userData);
+    });
+  }
+  handleSubmit(makeRequest, editProfilePopup, editProfilePopup.getForm());
```

Comment:
`.getForm()`


You need to pass `editFormValidator` here because you disable the button using a **validator**


```js

  handleSubmit(makeRequest, addCardPopup, editFormValidator);

```